tests/app/views/test_frameworks.py

```diff
@@ -845,6 +845,28 @@ def test_list_suppliers_by_multiple_statuses_2(self, live_g8_framework):
             assert len(data['supplierFrameworks']) == 3
             self.assert_supplier_ids(data, (3, 5, 6))
 
+    def test_list_suppliers_by_agreement_returned_true(self, live_g8_framework):
+        with self.app.app_context():
+            response = self.client.get(
+                '/frameworks/g-cloud-8/suppliers?agreement_returned=true'
+            )
+
+            assert response.status_code == 200
+            data = json.loads(response.get_data())
+            assert len(data['supplierFrameworks']) == 7
+            self.assert_supplier_ids(data, (3, 4, 5, 6, 7, 8, 9))
+
+    def test_list_suppliers_by_agreement_returned_false(self, live_g8_framework):
```

Comment:
These two new tests you've added are good but it looks to slightly duplicate tests on lines 577, 588. Maybe we could merge these together?

It looks like the test on 577 has an important assertion which could be absorbed into these new tests.
The test on 588 looks to not be doing much different and could probably be absorbed fully here?